### ISSUE TITLE
What is a difference with ubus over http?

### ISSUE THREAD

**Issue by stokito** [mkschreder/jucid#19]
I just found your project and still didn’t get it’s intention. Uhttpd has support for Ubus gateway so we can call any ubus service over http with JSON-RPC.
That’s looks very similar to what this project do. The only difference is that in orange you use websocjets and Lua.
Did I correctly got the point?
What is advantages or difference in comparison with Uhttpd +ubus?

**Comment by mkschreder** [comment]
In 2016 when I started JUCI there was luci2 which got some of the concepts
right and had a basic jsonrpc interface but was entirely unmanagable to
code against and not very useful. So I implemented what I considered
necessary at the time.

One of these things was the completely standalone orangerpcd rpc server
which does not need ubus but can be used to forward rpc calls to ubus
services. This allowed better access control and ability to not expose all
ubus services to the UI (I consider ubus to be for interprocess talk
primarily and not something to be exposed in its entirety over rpc
interface). It also allowed JUCI base services to be fairly standalone and
usable with only lua rpc calls that called shell scripts to communicate
with the system.

Also none of the ubus libraries are usable in a multithreaded application.
This was another reason to write a custom rpcd using code that is
reentrant.
